File: 49.Group_Anagrams.cs
public class Solution
{
    public IList<IList<string>> GroupAnagrams(string[] strs)
    {
        var d = new Dictionary<string, List<string>>();
        foreach (var str in strs)
        {
            var key = String.Concat(str.OrderBy(c => c));
            if (!d.ContainsKey(key))
            {
                d[key] = new List<string>();
            }
            d[key].Add(str);
        }
        return [.. d.Values];
    }
}

File: 49.Group_Anagrams.py
# sorting for transform [strings] to the [same string] and store them at the [same string] for those [strings]
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dict = defaultdict(list) # create a dict with default value is a list
        for str in strs:
            temp = ''.join(sorted(str))
            dict[temp].append(str)
        return list(dict.values())